security/auditing.rst

```diff
@@ -245,6 +245,80 @@ The table below contains some elements you can find in the generated AuditEvents
 +-----------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+
 
+
+.. _audit_event_signature:
+
+AuditEvent Signature
+--------------------
+
+An AditEvent Signature is a Provenance FHIR resource which contains a signature of the complete AuditEvent FHIR resource JSON. 
+This Provenance FHIR resource also includes a reference to an AuditEvent FHIR resource from which the signature is created. 
+
+.. note::
+
+   AuditEvent Signatures will not get generated if your configuration restricts the list of supported FHIR resources and ``Provenance`` is not included (see :ref:`supportedmodel`).
+
+AuditEvent Signature configuration
+----------------------------------
+
+By default generation of an AuditEvent Signature is disabled.
+To enable generation of an AuditEvent Signature add the following configuration to your Audit plugin configuration.
+
+.. code-block:: JavaScript
+
+   "Audit": {
+      "AuditEventSignatureEnabled": true, // Default is false
+      "AuditEventSignatureSecret": 
+      {
+          "SecretType": "JWKS", // Currently only supported type
+          // This is an example secret. Generate your own and do not use this example 'Secret' in your configuration!
+          "Secret": "{'keys':[{'kty':'EC','use':'sig','key_ops':['sign','verify'],'alg':'ES256','kid':'66e56ebf-a8de-4cfe-9710-3f2f44ec262f','crv':'P-256','x':'FO0bvAsRHC-wKMczT4xFPWQXI_fhFzqW2l9WxU29Hdc','y':'MHYht76KAnxHhatfB_BdyIuUtbpkK0g0Wuy5940oei4','d':'Nt1RXXNt6s5ytd88T7YhRePd7BqC4rh5WCOtJxdOzTs'}]}"
+      }
+    },
+
+Currenlty an ``AuditEventSignatureSecret`` can only contain a JSON Web Key Set ``Secret``.
+A JSON Web Key Set (JWKS) is a set of JSON Web Tokens (JWT) keys. 
+The JWKS is used for creating the signature of an AuditEvent.
+
+.. note::
+
+   Currently only the first key in a JSON Web Key Set is used to create signature of an AuditEvent.
+
+The following code snippet in C# is an example how you can generate a JSON Web Key Set. 
+
+.. code-block:: CSharp
+
+   using CreativeCode.JWK.KeyParts;
+   using CreativeCode.JWK;
+   
+   ...
+    
+   private static string CreateJSONWebKeySet()
+   {
+       var algorithm = Algorithm.ES256;
+       var keyUse = PublicKeyUse.Signature;
+       var keyOperations = new HashSet<KeyOperation>(new[] 
+                                { 
+                                    KeyOperation.ComputeDigitalSignature, 
+                                    KeyOperation.VerifyDigitalSignature 
+                                });
+       var jwk = new JWK(algorithm, keyUse, keyOperations);
+       var jwks = new JWKS(new[]{ jwk });
+
+       return jwks.Export();
+   }
```

Comment:
Could you maybe check whether there is a platform-independent shell/Powershell/terminal/etc version to generate this JWK(S)? Maybe something for node?..

Reply:
I added documentation to generate jwks with `jose`  command in a linux ( Ubuntu) terminal.